src/pages-and-resources/teams/Settings.jsx

```diff
@@ -0,0 +1,152 @@
+
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button, Form, Icon } from '@edx/paragon';
+import { Add } from '@edx/paragon/icons';
+import { TeamSetTypes, TeamSizes } from '../../data/constants';
+
+import FormikErrorFeedback from '../../generic/FormikErrorFeedback';
+import { setupYupExtensions, useAppSetting } from '../../utils';
+import AppSettingsModal from '../app-settings-modal/AppSettingsModal';
+import messages from './messages';
+import TeamSetEditor from './TeamSetEditor';
+
+setupYupExtensions();
+
+function TeamSettings({
+  intl,
+  onClose,
+}) {
+  const [teamsConfiguration, saveSettings] = useAppSetting('teamsConfiguration');
+  const blankNewTeamSet = {
+    name: '',
+    description: '',
+    type: TeamSetTypes.OPEN,
+    maxTeamSize: TeamSizes.DEFAULT,
+    id: null,
+    key: uuid(),
+  };
+
+  const handleSettingsSave = async (values) => {
+    // For newly-added teams, fill in an id.
+    const teamSets = values.topics?.map(teamSet => ({
+      id: teamSet.id || uuid(),
+      name: teamSet.name,
+      type: teamSet.type,
+      description: teamSet.description,
+      max_team_size: teamSet.maxTeamSize,
+    }));
+    return saveSettings({ topics: teamSets, max_team_size: values.maxTeamSize });
+  };
+
+  return (
+    <AppSettingsModal
+      appId="teams"
+      title={intl.formatMessage(messages.heading)}
+      enableAppHelp={intl.formatMessage(messages.enableTeamsHelp)}
+      enableAppLabel={intl.formatMessage(messages.enableTeamsLabel)}
+      learnMoreText={intl.formatMessage(messages.enableTeamsLink)}
+      onClose={onClose}
+      initialValues={{ ...teamsConfiguration }}
+      validationSchema={{
+        maxTeamSize: Yup.number()
+          .required(intl.formatMessage(messages.maxTeamSizeEmpty))
+          .min(TeamSizes.MIN)
+          .max(
+            TeamSizes.MAX,
+            intl.formatMessage(messages.maxTeamSizeTooHigh, {
+              max: TeamSizes.MAX,
+            }),
+          ),
+        topics: Yup.array().of(
+          Yup.object({
+            id: Yup.string().nullable(),
+            name: Yup.string()
+              .required(intl.formatMessage(messages.teamSetFormNameEmpty))
+              .trim(),
+            type: Yup.string().oneOf(Object.values(TeamSetTypes)),
+            description: Yup.string()
+              .required(intl.formatMessage(messages.teamSetFormDescriptionError))
+              .trim(),
+            maxTeamSize: Yup.number()
+              .required(intl.formatMessage(messages.maxTeamSizeEmpty))
+              .min(TeamSizes.MIN)
+              .max(
+                TeamSizes.MAX,
+                intl.formatMessage(messages.maxTeamSizeTooHigh, {
+                  max: TeamSizes.MAX,
+                }),
+              )
+              .default(null),
+          }),
+        )
+          .default([])
+          .uniqueProperty('name', intl.formatMessage(messages.teamSetFormNameExists)),
+      }}
```

Comment:
due to missing custom error messages for these validations, we are getting such errors on the frontend. have a look. 
<img width="749" alt="Screen Shot 2021-09-15 at 2 04 54 PM" src="https://user-images.githubusercontent.com/67791278/133408071-abb9ede5-1268-448f-ad1f-23315edb2fff.png">
<img width="480" alt="Screen Shot 2021-09-15 at 2 05 42 PM" src="https://user-images.githubusercontent.com/67791278/133408080-7d442b58-fed3-4b79-9ba4-ba6fb1e30932.png">

Reply:
I don't have a messages for these situations, but I've added a custom message for not. They can be updated with feedback from Jonathan.